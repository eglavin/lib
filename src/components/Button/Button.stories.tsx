import { Button, ButtonProps } from "./Button";
import type { Meta, StoryObj } from "@storybook/react";

const meta: Meta<typeof Button> = {
	component: Button,
};

export default meta;
type Story = StoryObj<typeof Button>;

export const Default: Story = {
	args: {
		label: "Button Text",
	},
	render: ({ ref, ...props }: ButtonProps) => <Button {...props} />,
};
