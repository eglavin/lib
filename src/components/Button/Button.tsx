import React, { forwardRef } from "react";
import "./Button.css";

export type ButtonProps = {
  label: string;
} & Omit<
  React.DetailedHTMLProps<
    React.ButtonHTMLAttributes<HTMLButtonElement>,
    HTMLButtonElement
  >,
  "children"
>;

export const Button = forwardRef<HTMLButtonElement, ButtonProps>(function Button(
  { label, ...props },
  ref
) {
  return (
    <button {...props} ref={ref}>
      {label}
    </button>
  );
});
