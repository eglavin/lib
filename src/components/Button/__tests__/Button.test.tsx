import { expect, describe, it } from "vitest";
import { render, screen } from "@testing-library/react";
import { Button } from "../Button";

describe("Button", () => {
	it("should render the Button component", () => {
		const { baseElement } = render(<Button label="Hello world!" />);

		expect(baseElement).toMatchSnapshot();
	});

	it("should have label content", () => {
		const TEST_LABEL = "Hello world!";

		render(<Button label={TEST_LABEL} />);

		expect(screen.getByRole("button", { name: TEST_LABEL })).toBeDefined();
	});
});
