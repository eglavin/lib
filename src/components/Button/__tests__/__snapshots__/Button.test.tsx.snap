// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should render the Button component 1`] = `
<body>
  <div>
    <button>
      Hello world!
    </button>
  </div>
</body>
`;
